test/util.ts

```diff
@@ -0,0 +1,119 @@
+import {Program} from "../src/runtime/dsl";
+import * as Runtime from "../src/runtime/runtime";
+import * as test from "tape";
+
+// You can specify changes as either [e,a,v] or [e,a,v,round,count];
+export type EAVArray = [Runtime.RawValue, Runtime.RawValue, Runtime.RawValue];
+export type EAVRCArray = [Runtime.RawValue, Runtime.RawValue, Runtime.RawValue, number, number];
+export type TestChange =  EAVArray | EAVRCArray;
+
+export function createChanges(transaction:number,eavns:TestChange[]) {
+  let changes:Runtime.Change[] = [];
+  for(let [e, a, v, round = 0, count = 1] of eavns as EAVRCArray[]) {
+    changes.push(Runtime.Change.fromValues(e, a, v, "my-awesome-node", transaction, round, count));
+  }
+  return changes;
+}
+
+export function verify(assert:any, program:Program, input:any[], output:any[], transaction = 1) {
+  let ins = createChanges(transaction, input);
+  let outs = createChanges(transaction, output);
+
+  let all = ins.concat(outs);
+  let {changes} = program.input(ins);
+  let msg = "Fewer changes than expected";
+  if(changes.length > all.length) {
+    msg = "More changes than expected";
+  }
+  assert.equal(changes.length, all.length, msg);
+
+  // Because the changes handed to us in expected aren't going to have the same
```

Comment:
NOTE: This algorithm is greedy. If we're convinced there is never a situation where two actual records might satisfy both expected records in one direction and only one in the other that's fine, otherwise this can get us into subtle bug territory. It is simpler than the previous system was though. I think it's fine for now, with a caveat emptor comment.